doc/contributing/hw/comportability/README.md

```diff
@@ -462,62 +462,137 @@ Here is an example specification:
+The determination of which input has interrupted the processor, sometimes referred to as "Disambiguation", is done primarily by querying the rv_plic claim/complete register (`CC`).
+This is distinct from another possible model in which each peripheral would send only one interrupt, and the processor would disambiguate by querying the peripheral to figure out which interrupt was triggered.
+
+### Register Creation
+
+By default for every peripheral, three registers are **automatically** created to manage the interrupts (as defined in the `interrupt_list` portion of the Hjson file).
+These registers are named `INTR_STATE`, `INTR_ENABLE` and `INTR_TEST`, and are placed at the top of the peripheral's address map in that order automatically by the `reggen` tool.
+Every interrupt has one bitfield in each of three registers.
+(It is currently an error if the `.hjson` defines more than 32 interrupts per-peripheral.)
+
+| Name          | Offset | Desc                                                                                  |
+|---------------|--------|---------------------------------------------------------------------------------------|
+| `INTR_STATE`  | 0x0    | holds the current state of the interrupt (may be RO or W1C depending on "IntrT").     |
+| `INTR_ENABLE` | 0x4    | enables/masks the output of INTR_STATE to create the wired interrupt signal `intr_o`. |
+| `INTR_TEST`   | 0x8    | write-only (`wo`) register which asserts the interrupt for testing purposes.          |
+
+The contents of the `INTR_STATE` register are not qualified by `INTR_ENABLE`, but rather show the raw state of all latched hardware interrupt events.
+
+### CIP Interrupt Types
+
+Interrupts are a mechanism for hardware to request attention from software, commonly as either as a normal part of a system's operation or as a way to handle exceptional circumstances.
+To design robust control flows and abstractions using interrupts, one way is to model Interrupts + Interrupt Handlers as forming a handshaking interaction between hardware and software.
+Hardware creates an interrupt request when a condition is true or an event occurs, and software responds to this request, possibly performing some action to address the condition/event, then acknowledges the interrupt request as completed.
+This request/acknowledge sequence is one-way to mitigate the effects of non-determinism in a real system, such as race-conditions due to the inherent round-trip latency from signalling to handling, and the unknown latency of a core executing the handler for one of possibly many competing interrupt requests.
+
+When creating an interrupt in a peripheral, we can identify two distinct behaviours a designer may choose to implement.
+
+1. **Event** type: Whenever an instantaneous event occurs trigger a new interrupt request, but do not create a new request until the previous request has been acknowledged (ack is a write to `INTR_STATE`, which is `W1C` in this mode).
```

Comment:
Grammar: I think you probably want "Whenever an ... event occurs, trigger ..."

(I genuinely misunderstood this at first!)

Reply:
Rephased, LMKWYT. I want to keep these two list items short and concise yet clear and accurate, which is tricky!

Reply:
I'm not sure this quite makes sense to me. At the moment, it's:
```
1. **Event** type: Whenever an instantaneous event occurs, trigger a new interrupt request.
   However, do not create a new request if the previous request has not yet been acknowledged (ack is a `W1C` operation to `INTR_STATE` in this mode).
```

Thinking a bit about the interrupt lines to the processor (which are level-sensitive), I'd expect one of the lines to be high until I acknowledge it. So I'm not sure I know how something *could* create a new request if I hadn't yet acknowledged it. (That might make sense if Ibex had edge-sensitive interrupts, but I don't think it does)

Reply:
I changed it to this:
```markdown
1. **Event** type: Whenever an instantaneous event occurs, trigger an interrupt request.
   The current interrupt can only be cleared by an acknowledgement from the handler (ack is a `W1C` operation to `INTR_STATE` in this mode).
```
Is that clearer to you? I think this is difficult to describe concisely, as the requesting/acknowledging concept I outlined in the previous paragraph struggles to mesh with how the interrupt is signalled over the wire (as you mention, with level-triggered interrupts new 'requests' could be viewed as impossible, or alternately as aliasing with previous requests on the bus). It depends on the context you view it from, as (looking just at the input signal) hardware has certainly tried to create a new request. I think it might make sense to flesh this distinction out further in the future.